client/starwhale/core/dataset/type.py

```diff
@@ -205,8 +205,10 @@ def __init__(
         mime_type: t.Optional[MIMEType] = None,
         dtype: t.Any = numpy.int8,
         encoding: str = "",
+        link: t.Optional[Link] = None,
     ) -> None:
         self.fp = str(fp) if isinstance(fp, Path) else fp
+        self._bytes: bytes = self.fp if isinstance(self.fp, bytes) else bytes()
```

Comment:
why store fp content twice?

Reply:
fp is designed to be a union variable, when calling to_bytes function, it can be translated into the content of the byte.

Reply:
can we remove the self._bytes property?

Reply:
> fp is designed to be a union variable, when calling to_bytes function, it can be translated into the content of the byte.

`self._bytes` could be considered as a private  cache content for this artifact. By contrast, fp is used as the initing variable for the class. `fp` is a union variable, when it is used as cache content, it is ambiguous.

Reply:
use `self.__cache_bytes` ?

Reply:
> use `self.__cache_bytes` ?

done